### PR TITLE
fix: postinstall script on Windows

### PR DESCRIPTION
**Pull request by williamluke4**

### CODE REVIEW

**Comment by williamluke4** [comment]
https://github.com/graphql-nexus/nexus-future/issues/525

**Comment by jasonkuhrt** [comment]
Will close https://github.com/graphql-nexus/nexus-future/issues/546

**Comment by williamluke4** [comment]
@jasonkuhrt Tested on windows and it works, could you test on your side?

File: scripts/postinstall.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+var mkdirp = require('mkdirp')
+var cpx = require("cpx");
```
Comment (jasonkuhrt) [inline scripts/postinstall.js]: Please use `fs-jetpack` as that's already a dep we use heavily

**Comment by jasonkuhrt** [comment]
> could you test on your side?

We'll wait to see how the e2e tests go once this is merged. 👍

File: scripts/postinstall.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const chalk = require('chalk')
+const fs = require('fs-extra')
+
+// proccess.env.PWD is undifined on Windows https://github.com/mrblueblue/gettext-loader/issues/18
+const pwd = process.cwd();
+const destDir = `${pwd}/../@types/nexus-plugin-prisma`
```
Comment (jasonkuhrt) [inline scripts/postinstall.js]: Surprised this works for windows, it doesn't need to be:

```suggestion
const destDir = `path.join(pwd, '..', '@types', 'nexus-plugin-prisma')`
```

?

File: scripts/postinstall.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+async function copyTypes() {
+  try {
+    await fs.ensureDir(destDir)
+    await fs.copy(`${pwd}/global-type.d.ts`, `${destDir}/index.d.ts`)
+  } catch (err) {
+    console.error(err)
```
Comment (jasonkuhrt) [inline scripts/postinstall.js]:
```suggestion
    console.error(err)
    process.exit(1)
```

Does what we were achieving with `&&` IIUC

File: scripts/postinstall.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+const chalk = require('chalk')
+const fs = require('fs-extra')
```
Comment (Weakky) [inline scripts/postinstall.js]: Any reason to use `fs-extra` instead of `fs-jetpack` which we already heavily use as mentioned by @jasonkuhrt ?

Comment (williamluke4) [reply scripts/postinstall.js]: @Weakky No reason just got confused

File: scripts/postinstall.js [hunk 5]
```diff
@@ -1,20 +1,18 @@
...
-async function copyTypes() {
-  try {
-    await fs.ensureDir(destDir)
-    await fs.copy(`${pwd}/global-type.d.ts`, `${destDir}/index.d.ts`)
-  } catch (err) {
-    console.error(err)
-    process.exit(1)
-  }
+try {
+  jetpack.dir(destDir)
```
Comment (jasonkuhrt) [inline scripts/postinstall.js]: Don't think this is needed with jetpack

File: scripts/postinstall.js [hunk 6]
```diff
@@ -1,20 +1,18 @@
...
+  jetpack.copy(`${pwd}/global-type.d.ts`, `${destDir}/index.d.ts`, { overwrite: true })
```
Comment (jasonkuhrt) [inline scripts/postinstall.js]: `path.join` again?

Comment (williamluke4) [reply scripts/postinstall.js]: Don't think it is necessary but iv changed it as it's a good practice

Comment (jasonkuhrt) [reply scripts/postinstall.js]: ok

File: scripts/postinstall.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+const chalk = require('chalk')
+const jetpack = require('fs-jetpack');
+const path = require('path');
+// proccess.env.PWD is undifined on Windows https://github.com/mrblueblue/gettext-loader/issues/18
+const pwd = process.cwd();
+const destDir = path.join(pwd, '..', '@types', 'nexus-plugin-prisma')
+
+jetpack.dir(destDir)
```
Comment (jasonkuhrt) [inline scripts/postinstall.js]:
```suggestion
```

?

**Comment by Weakky** [comment]
Merged! Thanks a lot @williamluke4 for your contribution 🎉🙏

**Pull request merged**